### ISSUE TITLE
At risk of shaving yaks, maybe CI to actually run this on PRs should also be included *with this PR*. A GitHub action exclusively for formatting shouldn't be terrifyingly complex, even if the CI will later be extended with probr & co.

### ISSUE THREAD

**Issue by peterrhysthomas** [finos/cloud-service-certification#117]
At risk of shaving yaks, maybe CI to actually run this on PRs should also be included *with this PR*. A GitHub action exclusively for formatting shouldn't be terrifyingly complex, even if the CI will later be extended with probr & co.

Otherwise it could make sense to postpone it until that lands; the cost of merging this PR won't increase since it's just autoformatting.

Slip-ups happen quickly, and reviewers probably won't pull PRs just to run `terraform fmt` on them, so IME having CI to check this is very important to stay consistent. There's also not much value to autoformatting unless it stays consistent (since it will just keep introducing whitespace changes when a mix of people using/not using it exists, often resulting in overhead on PRs).

_Originally posted by @TLATER in https://github.com/finos/cloud-service-certification/issues/104#issuecomment-815872809_

**Comment by peterrhysthomas** [comment]
Based on comments from @TLATER this issue is to build a GitHub action which operates on the PR and rejects the PR if the code is not already formatted by terraform fmt.  This has been moved to a new issue to distinguish this as a separate activity and so we can merge #104

**Comment by TLATER** [comment]
Can we retitle this to "Add CI for terraform fmt"? :)

**Comment by peterrhysthomas** [comment]
> Can we retitle this to "Add CI for terraform fmt"? :)

yes sorry I thought I did that already, but didn't press save obvs

**Comment by mcleo-d** [comment]
Hi @TLATER and @peterrhysthomas 👋 

I've changed to title to `Add CI for terraform fmt` as noted above.

James.

**Comment by TLATER** [comment]
This is part of the general work on #124 - we need to decide on the structure of our static checks before we can continue here.
